staging/src/k8s.io/apiserver/pkg/util/peerproxy/peer_discovery.go

```diff
@@ -0,0 +1,206 @@
+/*
+Copyright 2025 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package peerproxy
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"time"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/apiserver/pkg/authentication/user"
+	"k8s.io/client-go/discovery"
+	"k8s.io/klog/v2"
+
+	apidiscoveryv2 "k8s.io/api/apidiscovery/v2"
+	v1 "k8s.io/api/coordination/v1"
+	schema "k8s.io/apimachinery/pkg/runtime/schema"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	apirequest "k8s.io/apiserver/pkg/endpoints/request"
+	responsewriterutil "k8s.io/apiserver/pkg/util/responsewriter"
+)
+
+const (
+	controllerName = "peer-discovery-cache-sync"
+	maxRetries     = 5
+)
+
+func (h *peerProxyHandler) RunPeerDiscoveryCacheSync(ctx context.Context, workers int) {
+	defer utilruntime.HandleCrash()
+	defer h.peerLeaseQueue.ShutDown()
+	defer func() {
+		err := h.apiserverIdentityInformer.Informer().RemoveEventHandler(h.leaseRegistration)
+		if err != nil {
+			klog.Warning("error removing leaseInformer eventhandler")
+		}
+	}()
+
+	klog.Infof("Workers: %d", workers)
+	for i := 0; i < workers; i++ {
+		klog.Infof("Starting worker")
+		go wait.UntilWithContext(ctx, h.runWorker, time.Second)
+	}
+	<-ctx.Done()
+}
+
+func (h *peerProxyHandler) enqueueLease(lease *v1.Lease) {
+	h.peerLeaseQueue.Add(lease.Name)
```

Comment:
Realistically the queue key doesn't even matter now since we operate on the full state of all leases. @aojea is there a simpler way of setting up the queue or is this recommended?

Reply:
I use a constant "sync-token" or something like that as name so the queue aggregate all the events,